gems/sorbet-runtime/lib/types/props/private/serde_transform.rb

```diff
@@ -151,6 +154,35 @@ def self.generate(type, mode, varname)
         end
       end
 
+      sig {params(varname: String, type: T::Types::Base, mode: ModeType).returns(T.nilable(String)).checked(:never)}
+      private_class_method def self.handle_unknown_value_subtype(varname, type, mode)
+        inner = generate(type, mode, varname)
+        if inner.nil?
+          nil
+        else
+          case mode
+          when Serialize
+            <<~RUBY
+              if #{varname}.is_a?(T::Props::UnknownValue)
+                T::Props::Utils.deep_clone_object(#{varname}.serialized_value)
```

Comment:
Should we be building this on top of the `T::Props::CustomType` abstraction instead of special-casing something here?

For example, should `UnknownValue` implement `serialize` so that it could be serialized like any other value?

Reply:
Hmm... possibly. That could clean this up so we don't need the additional type check.